File: app/src/main/java/com/hungtran/bankingassistant/network/NetworkClient.java
package com.hungtran.bankingassistant.network;

import android.content.Context;

import com.hungtran.bankingassistant.util.Constant;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by hungtd on 2/18/19.
 */

public class NetworkClient {
    private static int TIME_OUT = 60;
    private static BankingApi bankingApi;
    private static OkHttpClient.Builder httpClient;
    private static Retrofit.Builder builder;
    private static HttpLoggingInterceptor httpLoggingInterceptor;
    private static Retrofit retrofit;

    public static BankingApi request() {
        builder = new Retrofit.Builder().baseUrl(Constant.API_BASE_URL).addConverterFactory(GsonConverterFactory.create());
        httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BASIC);
        httpClient = new OkHttpClient.Builder().connectTimeout(TIME_OUT, TimeUnit.SECONDS);
        httpClient.addInterceptor(httpLoggingInterceptor);
        retrofit = builder.client(httpClient.build()).build();
        bankingApi = retrofit.create(BankingApi.class);
        return  bankingApi;
    }
}

File: app/src/main/java/com/hungtran/bankingassistant/network/NetworkClient.kt
package com.hungtran.bankingassistant.network

import android.content.Context

import com.hungtran.bankingassistant.util.Constant

import java.util.concurrent.TimeUnit

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by hungtd on 2/18/19.
 */

object NetworkClient {
    private val TIME_OUT = 60
    lateinit var bankingApi: BankingApi
    lateinit var httpClient: OkHttpClient.Builder
    lateinit var builder: Retrofit.Builder
    lateinit var httpLoggingInterceptor: HttpLoggingInterceptor
    lateinit var retrofit: Retrofit

    fun request(): BankingApi {
        builder = Retrofit.Builder().baseUrl(Constant.API_BASE_URL).addConverterFactory(GsonConverterFactory.create())
        httpLoggingInterceptor = HttpLoggingInterceptor()
        httpLoggingInterceptor!!.level = HttpLoggingInterceptor.Level.BASIC
        httpClient = OkHttpClient.Builder().connectTimeout(TIME_OUT.toLong(), TimeUnit.SECONDS)
        httpClient!!.addInterceptor(httpLoggingInterceptor!!)
        retrofit = builder!!.client(httpClient!!.build()).build()
        bankingApi = retrofit!!.create(BankingApi::class.java)
        return bankingApi
    }
}
